pkg/controller/direct/bigquerydataset/bigquerydataset_mappings.go

```diff
@@ -21,210 +21,309 @@
 package bigquerydataset
 
 import (
+	"fmt"
 	"strconv"
+	"strings"
+	"time"
 
+	pb "cloud.google.com/go/bigquery"
 	krm "github.com/GoogleCloudPlatform/k8s-config-connector/apis/bigquery/v1beta1"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct"
-	api "google.golang.org/api/bigquery/v2"
 )
 
-func BigQueryDatasetSpec_ToAPI(mapCtx *direct.MapContext, in *krm.BigQueryDatasetSpec, name string) *api.Dataset {
+func BigQueryDatasetSpec_ToProtoMetadata(mapCtx *direct.MapContext, in *krm.BigQueryDatasetSpec) *pb.DatasetMetadata {
 	if in == nil {
 		return nil
 	}
-	out := &api.Dataset{}
-	acccessList := []*api.DatasetAccess{}
+	out := &pb.DatasetMetadata{}
+	acccessList := []*pb.AccessEntry{}
 	for _, access := range in.Access {
-		curAccess := Access_ToAPI(mapCtx, direct.LazyPtr(access))
+		curAccess := AccessEntry_ToProto(mapCtx, direct.LazyPtr(access))
 		acccessList = append(acccessList, curAccess)
 	}
 	out.Access = acccessList
 	out.DefaultCollation = direct.ValueOf(in.DefaultCollation)
-	out.DefaultPartitionExpirationMs = direct.ValueOf(in.DefaultPartitionExpirationMs)
-	out.DefaultTableExpirationMs = direct.ValueOf(in.DefaultTableExpirationMs)
-	out.DefaultEncryptionConfiguration = EncryptionConfiguration_ToAPI(mapCtx, in.DefaultEncryptionConfiguration)
+	out.DefaultPartitionExpiration = time.Duration(direct.ValueOf(in.DefaultPartitionExpirationMs)) * time.Millisecond
+	out.DefaultTableExpiration = time.Duration(direct.ValueOf(in.DefaultTableExpirationMs)) * time.Millisecond
+	out.DefaultEncryptionConfig = EncryptionConfiguration_ToProto(mapCtx, in.DefaultEncryptionConfiguration)
 	out.Description = direct.ValueOf(in.Description)
-	out.FriendlyName = direct.ValueOf(in.FriendlyName)
-	out.DatasetReference = DatasetReference_ToAPI(mapCtx, in, name)
+	out.ExternalDatasetReference = DatasetSpec_ToExternalDatasetReference(mapCtx, in)
```

Comment:
If currently we don't have this field in KRM go types, maybe we just remove this line for now. When we add this field in the future, we need to add the Go types first.